packages/babel-parser/src/parser/statement.js

```diff
@@ -200,12 +198,18 @@ export default class StatementParser extends ExpressionParser {
     if (nextCh === charCodes.leftCurlyBrace) return true;
 
     if (isIdentifierStart(nextCh)) {
-      let pos = next + 1;
-      while (isIdentifierChar(this.input.charCodeAt(pos))) {
-        ++pos;
+      keywordRelationalOperator.lastIndex = next;
+      const matched = keywordRelationalOperator.exec(this.input);
+      if (
+        matched !== null &&
+        // We have seen `in` or `instanceof` so far, now check if the identfier
+        // ends here
+        !isIdentifierChar(this.input.charCodeAt(next + matched[0].length))
```

Comment:
This is a bug, we should use `codePointAtPos` and also check the backslash
```
let in𝐬𝐭𝐚𝐧𝐜𝐞𝐨𝐟;
let in\u0061;
```
are both valid and currently throws. 

(I hate myself so much)